### PR TITLE
chore: migrate E2E uniswapTrade test to Foundry

### LINKED ISSUE #141: tests(e2e): `uniswapTrade`
May be implemented as a fork test.

### PR DESCRIPTION
**Pull request by meetmangukiya**
## Description

Migrate the uniswapTrade e2e test to foundry. Depends on https://github.com/cowprotocol/contracts/pull/215.

## Test Plan

CI

## Related Issues

Closes #141

### CODE REVIEW

File: test/e2e/UniswapTrade.t.sol [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+        assertEq(wETH.balanceOf(trader2.addr), 0.5 ether, "weth bought not correct amount");
+        assertEq(dai.balanceOf(trader2.addr), 300.3 ether - (uniswapDaiOutAmount + 0.3 ether));
+    }
+
+    function _getCode(string memory artifactName) internal view returns (bytes memory) {
+        string memory data =
+            vm.readFile(string(abi.encodePacked("node_modules/@uniswap/v2-core/build/", artifactName, ".json")));
+        return vm.parseJsonBytes(data, ".bytecode");
+    }
```
Comment (mfw78) [inline test/e2e/UniswapTrade.t.sol:188]: Dead code

File: test/e2e/UniswapTrade.t.sol [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    function setUp() public override {
+        super.setUp();
+
+        dai = deployMintableErc20("dai", "dai");
+        wETH = deployMintableErc20("wETH", "wETH");
+
+        factory = IUniswapV2Factory(0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f);
+        uniswapPair = IUniswapV2Pair(factory.createPair(address(wETH), address(dai)));
+
+        isWethToken0 = uniswapPair.token0() == address(wETH);
```
Comment (mfw78) [inline test/e2e/UniswapTrade.t.sol:48]: This is missing a comment from the test being migrated from.

File: test/e2e/UniswapTrade.t.sol [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+        assertEq(wETH.balanceOf(address(settlement)), 0.001 ether, "weth fees not as expected");
+        assertEq(dai.balanceOf(address(settlement)), 0.3 ether, "dai fees not as expected");
+
+        assertEq(wETH.balanceOf(trader1.addr), 0, "not all weth sold");
+        assertEq(dai.balanceOf(trader1.addr), uniswapDaiOutAmount * 2, "dai received not as expected");
+
+        assertEq(wETH.balanceOf(trader2.addr), 0.5 ether, "weth bought not correct amount");
+        assertEq(dai.balanceOf(trader2.addr), 300.3 ether - (uniswapDaiOutAmount + 0.3 ether));
+    }
```
Comment (mfw78) [inline test/e2e/UniswapTrade.t.sol:182]: Is there a reason why the uniswap reserve checks post-swap haven't been asserted?

File: test/e2e/UniswapTrade.t.sol (hunk 2 above)
Comment (meetmangukiya) [reply test/e2e/UniswapTrade.t.sol:48]: doesn't apply

**Review by mfw78 — APPROVED** [review]

**Comment by mfw78** [comment]
Merging with test CI/CD red due:
* Merging to an intermediate branch (not `main`)
* Issue is associated with a forked branch not having secret access to run the RPC.

**Pull request merged**